webapp/src/immersivereader.tsx

```diff
@@ -46,29 +60,98 @@ function beautifyText(content: string): string {
         );
     }
 
-    // Remove ** around bold text, replace with "
-    function cleanBold(content: string): string {
+    // If the ```block ``` annotation is used to display a block in
+    // a modal, remove it
+    function cleanHintText(content: string): string {
         return content.replace(
-            /\*\*([^*]+)\*\*/gu,
-            (matched, word, offset, s) => lf("\"{0}\"", word)
+            /^```block$[^`]*^```$/gum,
+            ""
+        );
+    }
+
+    // Remove ** and __ around bold text, replace with " if it's only one word
+    function cleanBold(content: string): string {
+
+        let singleWordBold = content.replace(
+            /\*\*([^*^\s]+)\*\*|__([^\s^_]+)__/gu,
+            (matched, boldStar, boldUnder, offset, s) => {
+                return "\"" + lf("{0}", (boldStar ? boldStar : boldUnder)) + "\""
+            }
+        )
+
+        return singleWordBold.replace(
+            /\*\*([^*]+)\*\*|__([^_]+)__/gu,
+            (matched, boldStar, boldUnder, offset, s) => {
+                return boldStar ? lf("{0}", boldStar) : lf ("{0}", boldUnder);
+            }
         );
     }
 
     // replace horizontal rules with new lines
     function cleanHorizontalRule(content: string): string {
         return content.replace(
-            /^\* \* \*$/gum,
-            "- - -"
+            /^\* \* \*$|^- - -$|^<hr\/>$/gum,
+            "<br /><br />"
         );
     }
 
     // replace consecutive new lines with break tags so that they don't get smooshed
     function cleanNewLines(content: string): string {
         return content.replace(
             /[\r\n]{2,}/gum,
-            `<br />`
+            `<br />\n`
         );
     }
+
+    // Remove >> for blockquotes
+    function cleanBlockquotes(content: string): string {
+        return content.replace(
+            /^>>/gu,
+            ""
+        )
+    }
+
+    // Remove all emojis because they get read out. Characters that aren't
+    // emojis don't get hit by this RegEx, but they're silent
+    function cleanEmojis(content: string): string {
```

Comment:
question: are we leaving the unicode characters (Ⓐ, Ⓑ, etc) for now?

Reply:
yesss because I don't think they get read out -- at least for the ones I've seen. Do you think it'd be better if it was replaced by the letters instead?

Reply:
yeah i think eventually the button ones should be replaced with "A" or "B" (since which button to select is important info for the game) but the other characters we probably don't need! could also add this later though, up to you